src/api/generic_api.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from config import Config
+import json
+import requests
+
+class GenericApi:
+    def __init__(self):
+        self._config = Config()
+
+    def send(self, msgData):
```

Comment:
I am not a massive fan of the msgData param where it is a dict and depending on the "type" it has different dict fields. Kind of error prone.

Reply:
Any other ways youd like to see? I'm struggling to think of a good one.  The goal was to keep everything going through messenger.send so the rest of botty just has to go through that function.  I guess thats kind of pointless if they need a type field though, so should the send message just be split out into a method for each of the different types?

Reply:
you could use something like @dataclass and then have a union of options. But I would be ok with the dict, just add a comment that explains the logic. So when a developer wants to use the function knows what to put into msgData.

Reply:
I went with breaking it down into separate functions.  I post an update with the interfaces and new functions.  Let me know what you think?

https://github.com/aeon0/botty/pull/370/commits/676ac1fb5f9a10608d3edd80c40e376f310e2cc8